pkg/gardenadm/botanist/botanist.go

```diff
@@ -57,6 +58,8 @@ type AutonomousBotanist struct {
 
 	// Bastion is only set for `gardenadm bootstrap`.
 	Bastion *bastion.Bastion
+	// ControlPlaneMachines is set by ListControlPlaneMachines during `gardenadm bootstrap`.
+	ControlPlaneMachines []machinev1alpha1.Machine
```

Comment:
Nit: should we unexport them? Looks like internal state

Reply:
It is internal state, but so are the components, and other members of the `(Autonomous)Botanist` structs. The flow or helper functions in other packages might still want to access them.

Reply:
I don't have a concrete usage for `ControlPlaneMachines` in mind, but for the `SSHConnection` in the next PR. I wanted to get rid of the local variables in the flow, which seemed weird to me :) 
Still, you're right. If needed, the member can simply be exposed via a method. I changed this in a new commit.